swupd/helpers_test.go

```diff
@@ -270,13 +295,30 @@ func checkFileInManifest(t *testing.T, m *Manifest, version uint32, name string)
 			if f.Version == version {
 				return
 			}
-			t.Errorf("in manifest %s version %d: file %s has version %d but expected %d", m.Name, m.Header.Version, f.Name, f.Version, version)
+			t.Errorf("in manifest %s version %d: file %s has version %d but expected %d",
+				m.Name, m.Header.Version, f.Name, f.Version, version)
 			return
 		}
 	}
 	t.Errorf("couldn't find file %s in manifest %s version %d", name, m.Name, m.Header.Version)
 }
 
+func fileInManifestHash(t *testing.T, m *Manifest, version uint32, name string, hash string) *File {
+	t.Helper()
+	for _, f := range m.Files {
+		if f.Name == name {
+			if f.Version != version {
+				t.Fatalf("in manifest %s version %d: file %s has version %d but expected %d", m.Name, m.Header.Version, f.Name, f.Version, version)
+			} else if f.Hash.String() != hash {
```

Comment:
logic seems weird here, why does the hash check depend on the version check preceding it? YMMV and this is just a style issue but it seems like this would read better as two separate conditionals instead of an `if/else if`

Reply:
Yes I'd like a comment on this block if possible.